### DISCUSSION TITLE
Related Work (Frameworks, Tools, etc)

### DISCUSSION THREAD

**Discussion by lfwa** [lfwa/carbontracker#50] [General]
Just wanted to add an overview of related work:
- [Scaphandre](https://github.com/hubblo-org/scaphandre): Track energy consumption of running processes (mostly CPU-based)
- [Boagent](https://github.com/Boavizta/boagent): Actively measure environmental impact of running applications (tries to include more life cycle phases, relies on Scaphandre)
- [CodeCarbon](https://github.com/mlco2/codecarbon/): Track carbon footprint from any kind of compute, not just ML, (also has a pretty dashboard)
- [experiment-impact-tracker](https://github.com/Breakend/experiment-impact-tracker): Track emissions of compute, focused on ML (no longer updated)
- [ML CO2 Calculator](https://mlco2.github.io/impact/): Calculate emissions from ML by inputting hardware used, hours used, provider, and region.

Feel free to expand this list.

**Comment by moonstarsky** [comment]
谢谢，您的邮件已收到~

**Comment by lfwa** [comment]
Update with newer works:
- [Website Carbon Calculator](https://github.com/ricardodantas/website-carbon-calculator): Calculates the carbon emission by the network traffic from a given URL
- [Google Cloud Region Picker](https://github.com/GoogleCloudPlatform/region-picker): ool to help you pick a Google Cloud region considering carbon footprint, price and latency
- [Carbonifer](https://github.com/carboniferio/carbonifer): Command Line Tool to control carbon emission of your cloud infrastructure
- [Tracarbon](https://github.com/fvaleye/tracarbon): Tracks your device's energy consumption and calculates your carbon emissions using your location (depends on [CO2signal](https://www.co2signal.com/) from [ElectricityMaps](https://www.electricitymaps.com/) and therefore requires users to sign up for an API key)
- [Cloud Carbon Footprint](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint): Estimates the energy (kilowatt hours) and carbon emissions (metric tons CO2e) of public cloud provider utilization
- [CATS](https://github.com/GreenScheduler/cats): Climate-Aware Task Scheduler. It schedules cluster jobs to minimize predicted carbon intensity of running the process
- [Green Algorithms](https://www.green-algorithms.org/): Web-tool and framework for clusters (see also [Green Algorithms for High Performance Computing](https://github.com/GreenAlgorithms/GreenAlgorithms4HPC))
